### PR TITLE
multiregionccl: add a cold start latency test 

### PR DESCRIPTION
**Pull request by ajwerner**
This commit adds a test which creates an MR serverless cluster and then
boots the sql pods in each region while disallowing connectivity to other
regions. It also simulates latency to make sure the routing logic works
and to provide a somewhat realistic picture of what to expect.

Epic: CRDB-18596

Release note: None

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
> Reviewed 2 of 2 files at r1, 3 of 3 files at r2, 1 of 1 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92826)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @JeffSwenson)

File: pkg/ccl/multiregionccl/cold_start_latency_test.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+		}
+		wg.Wait()
+		return latencyResults
+	}
+
+	enableLatency()
+	t.Log("pre running test to allocate instance IDs")
+	t.Log("result", localities, runAllTests())
+	t.Log("running test with no connectivity from sql pods to remote regions")
+	blockCrossRegionTenantAccess.Store(true)
```
Comment (JeffSwenson) [inline pkg/ccl/multiregionccl/cold_start_latency_test.go:369]: Blocking cross-region network traffic is a neat check.

**Review by jeffswenson — APPROVED** [review]
LGTM

**Review by ajwerner — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92826#-:-NMFvOqW9wlmj6fmEJqv:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/ccl/multiregionccl/cold_start_latency_test.go` line 71 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/92826#-NID2epR9ouP_S-CU_qI-r8-71:-NMFvDWhB7BEGthmcBCi:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e92630800ca507a1568b43b0447b952b3e771a7e/pkg/ccl/multiregionccl/cold_start_latency_test.go#L71)):*
<details><summary><i>Previously, JeffSwenson (Jeff Swenson) wrote…</i></summary><blockquote>

nit: is the extra block wrapping l.Find intentional?
</blockquote></details>

Done.

**Pull request merged**